### ISSUE TITLE
Search api score always 1

### ISSUE THREAD

**Issue by kangfenmao** [llm-tools/embedJs#186]
### 🐛 Describe the bug

![image](https://github.com/user-attachments/assets/321d2ab3-2043-4645-83fe-54c3d2c5dfc0)

LanceDB is ok

Libsql ragApplication.search score always is 1

![image](https://github.com/user-attachments/assets/f056ee5a-4907-4827-b214-74e03ef560b5)

**Comment by kangfenmao** [comment]
```
async similaritySearch(query, k) {
        const statement = `SELECT id, pageContent, uniqueLoaderId, source, metadata,
                vector_distance_cos(vector, vector32('[${query.join(',')}]')) as distance
            FROM ${this.tableName}
            ORDER BY vector_distance_cos(vector, vector32('[${query.join(',')}]')) ASC
            LIMIT ${k};`;
        this.debug(`Executing statement - ${truncateCenterString(statement, 700)}`);
        const results = await this.client.execute(statement);
        return results.rows.map((result) => {
            const metadata = JSON.parse(result.metadata.toString());
            console.debug('similaritySearch', JSON.stringify(result));
            return {
                metadata,
                pageContent: result.pageContent.toString(),
                score: result.distance,
            };
        });
    }
```

vector_distance_cos as distance

**Comment by kangfenmao** [comment]
Libsql query results are inaccurate

**Comment by kangfenmao** [comment]
score: 1 - result.distance

works

**Comment by kangfenmao** [comment]
https://github.com/kangfenmao/cherry-studio

**Comment by adhityan** [comment]
Thanks for pointing this out. This will be addressed in the next version.

**Comment by adhityan** [comment]
Resolved in `0.1.27`
